ocs_ci/ocs/amq.py

```diff
@@ -192,13 +247,227 @@ def setup_amq_kafka_bridge(self):
+            raise cf
+
+        # Making sure the producer pod is running
+        if self.is_amq_pod_running(pod_pattern="hello-world-producer"):
+            return self.producer_pod
+        else:
+            raise ResourceWrongStatusException("producer pod is not getting to running state")
+
+    def create_consumer_pod(self, num_of_pods=1, value='10000'):
+        """
+        Creates producer pods
+
+        Args:
+            num_of_pods (int): Number of consumer pods to be created
+            value (str): Number of messages to be received
+
+        Returns: consumer pod object
+
+        """
+        try:
+            consumer_pod = templating.load_yaml(constants.HELLO_WORLD_CONSUMER_YAML)
+            consumer_pod["spec"]["replicas"] = num_of_pods
+            consumer_pod["spec"]["template"]["spec"]["containers"][0]["env"][4]["value"] = value
+            self.consumer_pod = OCS(**consumer_pod)
+            self.consumer_pod.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during creation of consumer pod')
+            raise cf
+
+        # Making sure the producer pod is running
+        if self.is_amq_pod_running(pod_pattern="hello-world-consumer"):
+            return self.consumer_pod
+        else:
+            raise ResourceWrongStatusException("consumer pod is not getting to running state")
+
+    def validate_messages_are_produced(self, value='10000', since_time=1800):
+        """
+        Validates if all messages are sent in producer pod
+
+        Args:
+            value (str): Number of messages are sent
+            since_time (int): Number of seconds to required to sent the msg
+
+        Returns:
+            bool: True if all messaged are sent in producer pod.
+
+        """
+        _rc = True
+        producer_pod_objs = [get_pod_obj(
+            pod
+        )for pod in get_pod_name_by_pattern('hello-world-produce', self.namespace)
+        ]
+        for pod in producer_pod_objs:
+            cmd = f"oc logs -n {self.namespace} {pod.name} --since={since_time}s"
+            msg = run_cmd(cmd)
+            if msg.find(f"{value} messages sent") is -1:
+                _rc = False
+                log.error(f"On producer {pod.name} all or few messages are not sent")
+        return _rc
+
+    def validate_messages_are_consumed(self, value='10000', since_time=1800):
+        """
+        Validates if all messages are received in consumer pod
+
+        Args:
+            value (str): Number of messages are recieved
+            since_time (int): Number of seconds to required to receive the msg
+
+        Returns:
+            bool: True if all messaged are received in consumer pod.
+
+        """
+        _rc = True
+        producer_pod_objs = [get_pod_obj(
+            pod
+        )for pod in get_pod_name_by_pattern('hello-world-consumer', self.namespace)
+        ]
+        for pod in producer_pod_objs:
```

Comment:
I see some error here in the logs you attached.
11:56:36 - MainThread - ocs_ci.ocs.amq - ERROR - On consumer hello-world-consumer-66d8cc5c57-qmhvc all or few messages are not sent.
Do you think we will have to raise some exception here instead of logging as error?

Reply:
yes it is better to raise exception. Will address this in next commit